File: Utils/XmlDocumentLoader.cs
using System.Xml;

namespace XmlEditorUi;

public static class XmlDocumentLoader
{
    public static XmlDocument LoadFromFile(string path)
    {
        var doc = new XmlDocument { PreserveWhitespace = true };
        doc.Load(path);
        return doc;
    }
}

File: flutter/lib/data/xml/xml_document_loader.dart
import 'dart:io';

import 'package:xml/xml.dart';

import 'xml_file_io.dart';

class XmlDocumentLoader {
  static XmlDocument loadFromFile(String path) => XmlFileIo.loadDocument(path);

  static void saveToFile(XmlDocument document, String path, {bool indent = true}) {
    final file = File(path);
    file.parent.createSync(recursive: true);
    if (indent) {
      file.writeAsStringSync(document.toXmlString(pretty: true, indent: '    '));
    } else {
      file.writeAsStringSync(document.toXmlString());
    }
  }
}
